packages/node-cli/src/core/CommandActionExecutor.js

```diff
@@ -83,20 +83,32 @@ module.exports = class CommandActionExecutor {
 				projectConfiguration: projectConfiguration,
 			});
 
-			this._commandOutputHandler.showSuccessResult(actionResult, command.formatOutputFunc);
+			if (!(actionResult instanceof ActionResult)) {
+				throw 'INTERNAL ERROR: Command must return an ActionResult object.';
+			}
 
-			if (actionResult && actionResult.operationResult) {
-				const operationResult = actionResult.operationResult;
-				if (operationResult.status === OperationResultStatus.SUCCESS && commandUserExtension.onCompleted) {
-					commandUserExtension.onCompleted(actionResult);
-				} else if (operationResult.status === OperationResultStatus.ERROR && commandUserExtension.onError) {
-					const error = SDKOperationResultUtils.getResultMessage(operationResult)
-						+ NodeUtils.lineBreak
-						+ SDKOperationResultUtils.getErrorMessagesString(operationResult);
-					commandUserExtension.onError(error);
+			if (actionResult.status === ActionResult.ERROR) {
+				if (actionResult.errorMessages && Array.isArray(actionResult.errorMessages) && actionResult.size > 0) {
```

Comment:
I think last condition should be actionResult.errorMessages.length>0.
Anyway we have a method that is taking care of getting the errorMessages including the resultMessage.
So we can:
```
if (actionResult.status === ActionResult.ERROR) {
   throw ActionResultUtils.getErrorMessagesString(actionResult);
}
```

Reply:
Nice!